check-mysql/lib/check-mysql.go

```diff
@@ -52,12 +62,60 @@ SubCommands:`)
 	ckr.Exit()
 }
 
-func newMySQL(m mysqlSetting) mysql.Conn {
+func newDB(m mysqlSetting) (*sql.DB, error) {
 	proto := "tcp"
 	target := fmt.Sprintf("%s:%s", m.Host, m.Port)
 	if m.Socket != "" {
 		proto = "unix"
 		target = m.Socket
 	}
-	return mysql.New(proto, "", target, m.User, m.Pass, "")
+	cfg := &mysql.Config{
+		User:                 m.User,
+		Passwd:               m.Pass,
+		Net:                  proto,
+		Addr:                 target,
+		AllowNativePasswords: true,
+	}
+
+	return sql.Open("mysql", cfg.FormatDSN())
+}
+
+func getMySQLVersion(db *sql.DB) (*mysqlVersion, error) {
```

Comment:
[offtopic]
Maybe the best way to retrieve MySQL version is [`mysql_get_server_version()` C API](https://dev.mysql.com/doc/c-api/8.0/en/mysql-get-server-version.html), but maybe it's not available in go-sql-driver/mysql, maybe…

Reply:
Yes, go-sql-driver/mysql does not have such API.